service/history/replicatorQueueProcessor.go

```diff
@@ -451,6 +455,7 @@ func (p *replicatorQueueProcessorImpl) getTasks(
 	var replicationTasks []*replicator.ReplicationTask
 	readLevel := lastReadTaskID
 	for _, taskInfo := range taskInfoList {
+		_ = p.rateLimiter.Wait(ctx)
```

Comment:
if context timedout during waiting for token, should we just abort and return an error?

Reply:
It will return error from the below calls